include/language.php

```diff
@@ -571,7 +571,12 @@ public function to_array( $context = 'display' ) {
 	 * @return stdClass Converted `PLL_Language` object.
 	 */
 	public function to_std_class() {
-		return (object) $this->to_array();
+		$object = (object) $this->to_array();
+
+		// Sets the w3c locale as the main locale.
+		$object->locale = $this->get_locale( 'display' );
```

Comment:
Please not here. There is a risk that we need to convert the object outside the case you are handling today.
This should be done in `PLL_Walker` as discussed.

Reply:
See https://github.com/polylang/polylang/pull/1529/commits/1654f17ed8a33f0af294b86bbb81e820377ec796